### PR TITLE
Fix error message on row dereference

### PR DESCRIPTION
**Pull request by wendigo**
Fixes https://github.com/trinodb/trino/issues/23363

Release notes: improve error message while dereferencing an unknown row field

### CODE REVIEW

**Review by martint — APPROVED** [review]

File: core/trino-main/src/test/java/io/trino/sql/analyzer/TestAnalyzer.java [hunk 1]
```diff
@@ -384,6 +384,14 @@ public void testHavingReferencesOutputAlias()
                 .hasMessage("line 1:32: Column 'x' cannot be resolved");
     }
 
+    @Test
+    public void testRowReferencesUnknownField()
+    {
+        assertFails("SELECT row('a', 'b', 'c').field")
+                .hasErrorCode(COLUMN_NOT_FOUND)
+                .hasMessage("line 1:8: Column 'field' cannot be resolved");
```
Comment (findepi) [inline core/trino-main/src/test/java/io/trino/sql/analyzer/TestAnalyzer.java:392]: "Column field" or "Field field" ?

Comment (wendigo) [reply core/trino-main/src/test/java/io/trino/sql/analyzer/TestAnalyzer.java]: PTAL now

File: core/trino-main/src/test/java/io/trino/sql/analyzer/TestAnalyzer.java [hunk 2]
```diff
@@ -384,6 +384,14 @@ public void testHavingReferencesOutputAlias()
                 .hasMessage("line 1:32: Column 'x' cannot be resolved");
     }
 
+    @Test
+    public void testRowReferencesUnknownField()
+    {
+        assertFails("SELECT row('a', 'b', 'c').field")
+                .hasErrorCode(INVALID_COLUMN_REFERENCE)
```
Comment (losipiuk) [inline core/trino-main/src/test/java/io/trino/sql/analyzer/TestAnalyzer.java:391]: INVALID_COLUMN_REFERENCE seems wrong. Do we need new error code.
cc: @martint

Comment (wendigo) [reply core/trino-main/src/test/java/io/trino/sql/analyzer/TestAnalyzer.java]: I've added a new error type. @martint ptal

Comment (martint) [reply core/trino-main/src/test/java/io/trino/sql/analyzer/TestAnalyzer.java]: The INVALID_COLUMN_REFERENCE error code is fine for this purpose. Fields and columns are two ways to look at the same thing. Precisely, a table is a `multiset` of values of type `row`, so top level columns *are* fields.

Comment (wendigo) [reply core/trino-main/src/test/java/io/trino/sql/analyzer/TestAnalyzer.java]: Will revert then

**Pull request merged**